src/goto-instrument/accelerate/acceleration_utils.cpp

```diff
@@ -887,7 +886,7 @@ bool acceleration_utilst::do_nonrecursive(
   // for these variables by just forward simulating the path and
   // taking the expressions we get at the end.
   replace_mapt state;
-  expr_sett array_writes;
+  std::unordered_set<index_exprt, irep_hash> array_writes;
```

Comment:
Unrelated change? (There's probably a lot of cleanup to be done in this code, and it's not a big deal to include this here, but why only `array_writes` and not the other two below?)

Reply:
It's related: to access opX safely, you need to know that there are two operands, which is the case because that set contains `index_exprt` expressions. This invariant is now made explicit.